remotes/docker/resolver.go

```diff
@@ -283,8 +283,12 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 			log.G(ctx).Debug("resolving")
 			resp, err := req.doWithRetries(ctx, nil)
 			if err != nil {
+				if lastErr == nil {
```

Comment:
What's the intention behind adding this logic? It's duplicated logic of the code further down on line 294.